src/openms/include/OpenMS/ANALYSIS/XLMS/OpenProXLUtils.h

```diff
@@ -0,0 +1,302 @@
+        NUMBER_OF_CROSS_LINK_TYPES
+      };
+
+      AASequence alpha; // longer peptide
+      AASequence beta; // shorter peptide (empty for mono-link), tie bracker: mass then lexicographical
+      std::pair<SignedSize, SignedSize> cross_link_position; // index in alpha, beta or between alpha, alpha in loop-links
+      double cross_linker_mass;
+      String cross_linker_name;
+      ResidueModification::TermSpecificity term_spec_alpha;
+      ResidueModification::TermSpecificity term_spec_beta;
+
+      ProteinProteinCrossLinkType getType() const
+      {
+        if (!beta.empty()) return CROSS;
+
+        if (cross_link_position.second == -1) return MONO;
+
+        return LOOP;
+      }
+
+    };
+
+struct CrossLinkSpectrumMatch
+  {
+    /// structure of the cross-link
+    ProteinProteinCrossLink cross_link;
+
+    /// reference to pair of spectra
+    Size scan_index_light;
+    Size scan_index_heavy;
+
+    /// final score
+    double score;
+
+    /// rank among the matches to the same spectrum
+    Size rank;
+
+    /// counts, scores and other data for xQuest-like output
+    double pre_score;
+    double percTIC;
+    double wTIC;
+    double int_sum;
+    double match_odds;
+    std::vector< double > xcorrx;
+    double xcorrx_max;
+    std::vector< double > xcorrc;
+    double xcorrc_max;
+    Size matched_common_alpha;
+    Size matched_common_beta;
+    Size matched_xlink_alpha;
+    Size matched_xlink_beta;
+    double HyperCommon;
+    double HyperXlink;
+    double HyperAlpha;
+    double HyperBeta;
+    double HyperBoth;
+    double PScoreCommon;
+    double PScoreXlink;
+    double PScoreAlpha;
+    double PScoreBeta;
+    double PScoreBoth;
+
+    std::vector<PeptideHit::FragmentAnnotation> frag_annotations;
+
+    Size peptide_id_index;
+//    PeptideIdentification *peptide_id = NULL;
+
+    bool operator<(const CrossLinkSpectrumMatch& other) const
+    {
+      return score < other.score;
+    }
+
+    bool operator==(const CrossLinkSpectrumMatch& other) const
+    {
+      return cross_link.alpha == other.cross_link.alpha &&
+           cross_link.beta == other.cross_link.beta &&
+           cross_link.cross_link_position == other.cross_link.cross_link_position &&
+           score == other.score &&
+           pre_score == other.pre_score &&
+           percTIC == other.percTIC &&
+           wTIC == other.wTIC &&
+           int_sum == other.int_sum &&
+           match_odds == other.match_odds &&
+           xcorrx == other.xcorrx &&
+           xcorrx_max == other.xcorrx_max &&
+           xcorrc == other.xcorrc &&
+           xcorrc_max == other.xcorrc_max &&
+           matched_common_alpha == other.matched_common_alpha &&
+           matched_common_beta == other.matched_common_beta &&
+           matched_xlink_alpha == other.matched_xlink_alpha &&
+           matched_xlink_beta == other.matched_xlink_beta &&
+           rank == other.rank;
+    }
+
+  };
+
+
+  struct OPENMS_DLLAPI OpenProXLUtils
```

Comment:
please use `class` for this instead of `struct` and only expose public functions where necessary. Also consider changing the class name to something more appropriate than `OpenProXLUtils`. 

I am a bit afraid that this is just a catch-all class that tries to do *everything* and just contains a whole bunch of static functions without any documentation. There is no public interface and description what the class is supposed to do and which functions should be called from the outside. Also none of the functions have a doxygen-string.

I thus highly recommend to break up this class into useful sub-classes that are functionally related. Also I would suggest to create a public interface that only has a few functions and that will be called by the UTIL. One class could be `OpenMS/ANALYSIS/XLMS/OpenProXLScoring.h` , another one  `OpenMS/ANALYSIS/XLMS/OpenProXLSpectralAlignment.h` and then  `OpenMS/ANALYSIS/XLMS/OpenProXLSpectralDecharger.h` another one could be `OpenMS/ANALYSIS/XLMS/OpenProXLFileFormat.h` (or so) and of course the obvious  `OpenMS/ANALYSIS/XLMS/OpenProXLHelper.h` ...